source/com.microsoft.tfs.client.common/src/com/microsoft/tfs/client/common/credentials/CredentialsHelper.java

```diff
@@ -224,7 +228,7 @@ private static void removeStaleOAuth2Token() {
         if (oauth2TokenPair != null && oauth2TokenPair.AccessToken != null) {
             final String token = oauth2TokenPair.AccessToken.Value;
 
-            if (!StringUtil.isNullOrEmpty(token) && !isOAuth2TokenValid(token)) {
+            if (force || !StringUtil.isNullOrEmpty(token) && !isOAuth2TokenValid(token)) {
```

Comment:
could you add nested parenthesis to make it more clear?  I believe the order of operations is ( (force || !isNull) && isValid) but seeing the name force makes me think you meant (force || (!isNull && !isValid)